src/ir/table-utils.h

```diff
@@ -84,7 +84,13 @@ inline Index append(Table& table, Name name, Module& wasm) {
     }
     wasm.dylinkSection->tableSize++;
   }
-  segment->data.push_back(name);
+
+  auto* func = wasm.getFunctionOrNull(name);
+  if (func == nullptr) {
+    Fatal() << "Cannot append non-existing function to a table.";
```

Comment:
I needed to safely get the function signature, otherwise it would crash in case of an undefined function before reaching the validator. Maybe there's a better way to do that?

Reply:
I think the code is correct. Yes, you need the function to exist to know the signature. But the `Fatal()` should be an assert.